extensions/git/src/repository.ts

```diff
@@ -1247,40 +1254,65 @@ export class Repository implements Disposable {
 	}
 
 	async commit(message: string | undefined, opts: CommitOptions = Object.create(null)): Promise<void> {
-		const indexResources = [...this.indexGroup.resourceStates.map(r => r.resourceUri.fsPath)];
-		const workingGroupResources = opts.all && opts.all !== 'tracked' ?
-			[...this.workingTreeGroup.resourceStates.map(r => r.resourceUri.fsPath)] : [];
+		const clearInputBoxAndCloseDiffEditors = async () => {
+			if (message) {
+				this.inputBox.value = await this.getInputTemplate();
+			}
+
+			const indexResources = [...this.indexGroup.resourceStates.map(r => r.resourceUri.fsPath)];
+			const workingGroupResources = opts.all && opts.all !== 'tracked' ?
+				[...this.workingTreeGroup.resourceStates.map(r => r.resourceUri.fsPath)] : [];
+
+			this.closeDiffEditors(indexResources, workingGroupResources);
+		};
```

Comment:
Why create this temporary function for this _post commit stuff_ when we can run all of this after the `if else` clause?

Reply:
@joaomoreno, not sure which `if else` you have in mind. I moved this into a temporary function as this has to be executed in both the `RebaseContinue` and `Commit` operations.

Reply:
Oh I see, you want all that action to happen as part of the operation. Then I suggest creating a `private` method in the class instead of creating a temporary function every time this runs.